lms/djangoapps/course_api/tests/test_api.py

```diff
@@ -0,0 +1,185 @@
+"""
+Test for course API
+"""
+
+from django.contrib.auth.models import AnonymousUser
+from django.http import Http404
+from django.test import RequestFactory
+from rest_framework.exceptions import PermissionDenied
+
+from opaque_keys.edx.keys import CourseKey
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase, ModuleStoreTestCase
+
+from ..api import course_detail, list_courses
+from .mixins import CourseApiFactoryMixin
+
+
+class CourseApiTestMixin(CourseApiFactoryMixin):
+    """
+    Establish basic functionality for Course API tests
+    """
+
+    maxDiff = 5000  # long enough to show mismatched dicts
+
+    expected_course_data = {
+        'course_id': u'edX/toy/2012_Fall',
+        'name': u'Toy Course',
+        'number': u'toy',
+        'org': u'edX',
+        'short_description': u'A course about toys.',
+        'media': {
+            'image': {
+                'uri': u'/c4x/edX/toy/asset/just_a_test.jpg',
+                'name': u'Course Image',
+                'description': u'',
+                'type': u'logo',
+            }
+        },
+        'start': u'2015-07-17T12:00:00Z',
+        'start_type': u'timestamp',
+        'start_display': u'July 17, 2015',
+        'end': u'2015-09-19T18:00:00Z',
+        'enrollment_start': u'2015-06-15T00:00:00Z',
+        'enrollment_end': u'2015-07-15T00:00:00Z',
+        'blocks_url': '/api/courses/v1/blocks/?course_id=edX%2Ftoy%2F2012_Fall',
+    }
+
+    @classmethod
+    def setUpClass(cls):
+        super(CourseApiTestMixin, cls).setUpClass()
+        cls.request_factory = RequestFactory()
```

Comment:
Not really.  I've always done it that way, because it seemed to make sense to create one factory to use for all the requests we want to make.  Do you prefer it at the setUp level?